Docs 3303: Restructure ready-made integration manuals (#2135)
MultiSafepay/docs: content/integration/ready-made/magento2/_index.md

The lifetime of bank transfers is 86400 minutes (60 days).

The order status in Magento 2 changes to **Cancelled** before the payment can be matched to the order.

{{< /details >}}

### Payment components

The plugin supports [Payment Components](/payment-components/), which:

- Provide a seamless checkout experience to increase conversion.
- Encrypt customer payment details for secure processing.
- Shift responsibility for [PCI DSS compliance](/glossaries/multisafepay-glossary/#payment-card-industry-data-security-standard-pci-dss) to MultiSafepay.

{{< details title="Activating the payment component in your backend" >}}

1. Sign in to your Magento 2 backend.
2. Go to **Configuration** > **Credit card**, and then click **Configure**.
3. Click **Payment type** and select **Credit card component**.
4. Click **Save config**.

For questions, email <integration@multisafepay.com>

**Note:** If you have a custom checkout and encounter a conflict with the payment component, the Integration Team will do their best to provide support, but we can't guarantee compatibility in all cases.

{{< /details >}}

### Payment links 

> From version 2.0 and higher, [payment links](/payment-links/) are automatically generated for every order.

Author: ```suggestion
From version 2.0 and higher, [payment links](/account/payment-links/) are automatically generated for every order.
```

php bin/magento setup:static-content:deploy
```

**2.** To enable all modules, run in the Magento 2 root directory:
```
./bin/magento module:enable `./bin/magento module:status | grep MultiSafepay_`
```

### Stock handling

When you disable MSI in Magento 2, you must also disable the MultiSafepay MSI module by running:
```
php bin/magento module:disable MultiSafepay_ConnectMSI
```
If you have enabled MSI in Magento 2, to disable the MultiSafepay CatalogInventory module, run:
```
php bin/magento module:disable MultiSafepay_ConnectCatalogInventory
```

## Configuration
1. Sign in to your Magento 2 backend.
2. Go to **Stores** > **Configuration** > **MultiSafepay**.  

{{< details title="Specific settings" >}}

> This section contains the following pages:

Reviewer: I'm not clear what this means. The configuration section doesn't have the mentioned sections below it?

> This section contains the following pages:

Author: ```suggestion
```

{{< /details >}}

### Generic gateways

The plugin supports generic gateways, which redirect customers from your checkout to a MultiSafepay [payment page](/payment-pages/). This is particularly useful for integrating gift cards.

{{< details title="Configuring generic gateways" >}}

1. Sign in to your Magento 2 backend.
2. Go to **Stores** > **Configuration** > **MultiSafepay** > **Payment methods** > **Generic gateway**.
3. Set the relevant [payment method gateway IDs](https://docs-api.multisafepay.com/reference/gateway-ids) and upload a custom gateway image.
4. For pay later methods, specify whether to include a shopping cart.

For support, email <integration@multisafepay.com>

{{< /details >}}

### Logs

{{< details title="Downloading MultiSafepay logs" >}}

1. Sign in to your Magento backend.
2. Go to **Stores** > **Configuration** > **MultiSafepay** > **General settings** > **MultiSafepay payment logs**.
3. Click **Download**.

You receive a ZIP package containing a system report file and any MultiSafepay log(s) stored in the `/var/log` directory.

{{< /details >}}

### Magento Vault and Instant Purchase

> Magento Vault enables you to use [Instant purchase](https://magento.com/innovations-lab/instant-purchase), a feature that helps make repeat payments faster and easier, increasing conversion. 

Reviewer: ```suggestion
Magento Vault enables you to use [Instant purchase](https://magento.com/innovations-lab/instant-purchase), a feature that helps make repeat payments faster and easier, increasing your conversion. 
```

- Inform customers how they can delete stored payment details.

{{< /details >}}

{{< details title="Activating Magento Vault" >}} 
&nbsp;  
To activate Magento Vault, email a request to enable [recurring payments](/features/recurring-payments/) to <sales@multisafepay.com>

{{< /details >}}

{{< details title="Vault security" >}} 

If you host Magento yourself, the security of the vault depends on the security of your server. 

However, the vault only contains tokens valid in your webshop. If your server is compromised, no actual payment details are at risk, only stored customer data.   

{{< /details >}}

### Order lifetimes

The default lifetime of **Pending payment** orders in Magento 2 is 480 minutes (8&nbsp;hours). For payment methods with a longer authorization period, the order status changes to **Cancelled** after 8 hours.

{{< details title="Extending order lifetimes" >}}

To extend the lifetime of pending payments orders, increase the **Order Cron settings** value to longer than the validation period.

For instructions, see Magento – [Pending payment order lifetime](https://docs.magento.com/user-guide/sales/order-pending-payment-lifetime.html).

**ERP systems**

> For ERP systems, if the order status is **Declined**, successful payments often fail to process for orders with **Cancelled** status.

Reviewer: Should this be in this section?

5. Under **Notes for this order**, copy the payment link.
6. Send the payment link to the customer.

{{< /details >}}

{{< details title="Adding payment links to order confirmation emails" >}}

1. Sign in to your Magento 2 backend.
2. Go to **Marketing** > **Email templates**.
3. Under **New order**, import a template.
4. Add the following code snippet to the template:

```
{{depend order.getPayment().getAdditionalInformation('payment_link')}}
<a href="{{var order.getPayment().getAdditionalInformation('payment_link')}}">Pay now with {{var order.getPayment().getAdditionalInformation('method_title')}}</a>
{{/depend}}
```

**Magento 2.3.4+**

The email template syntax is different for Magento 2.3.4+. Add this code snippet instead:

```
{{depend order.payment.additional_information.payment_link}}
<a href="{{var order.payment.additional_information.payment_link}}">Pay now with {{var order.payment.additional_information.method_title}}</a>
{{/depend}}
```

**Backend emails**

To add payment links to order confirmation emails from your Magento **backend**, you can use the `payment_link` variable and an `if/else` statement in the template. 

**Frontend emails**

> You cannot add payment links to order confirmation emails created in your **frontend**, but you can dispaly the name of the payment method:

Reviewer: ```suggestion
You cannot add payment links to order confirmation emails created in your **frontend**, but you can display the name of the payment method:
```

`{{if payment_link}}`
`<a href="{{var payment_link}}">Pay now with {{var order.payment.additional_information.method_title}}</a>`
`{{else}}`
`<p>{{var order.payment.additional_information.method_title}}</p>`
`{{/if}}`

1. Go to **Stores** > **Configuration** > **Sales** > **Sales emails**.
2. Replace the **New order confirmation template** with your template.
3. Test the template to confirm it is working.

**Note:** You can also implement this directly in the email template files.

{{< /details >}}

### Payment methods

> {{< details title="Payment methods" >}}

Reviewer: ```suggestion
{{< details title="Supported payment methods" >}}
```

    - Boekenbon
    - Edenred
    - Fashioncheque
    - Fashion gift card
    - Fietsenbon
    - Gezondheidsbon
    - Givacard
    - Good4fun
    - Goodcard
    - Nationale tuinbon
    - Paysafecard
    - Parfumcadeaukaart
    - Podium
    - Sport en Fit
    - VVV gift card
    - Webshop gift card
    - Wellness gift card
    - Wijncadeau
    - Winkelcheque
    - Yourgift

See also [MultiSafepay gateway](/developer/generic-gateways/#multisafepay-gateways).

{{< /details >}}

### Progressive web apps

The plugin is compatible with GraphQL queries and can be integrated into PWA stores using an additional [GraphQL support module](https://github.com/MultiSafepay/magento2-graphql).

We also offer full extensions for [ScandiPWA](/scandipwa/) and [Vue Storefront](/vue-storefront/). 

### Refunds

{{< details title="Refund rules" >}}
|  |   |
|---|---|
| MultiSafepay dashboard | - Full and partial refunds <br> - Orders with [Fooman surcharges](/payments/integrations/ecommerce-platforms/magento2/faq/applying-surcharges/) <br> - Orders from the deprecated plugin |
| Backend | - Full and partial refunds, and credit memos <br> - You can't refund more than the original amount in your backend |

> | API | - See [Refund order](https://docs-api.multisafepay.com/reference/refundorder) > Pay later refund <br> - PATCH requests **not** supported |

Reviewer: ```suggestion
| API | - See [Refund order](https://docs-api.multisafepay.com/reference/refundorder) > Pay later refund <br> - PATCH requests are **not** supported |
```

3. Click the relevant refund button:  
    - **Offline refund**: Doesn't send a refund request to MultiSafepay.
    - **Refund**: Sends a refund request to MultiSafepay.

{{< /details >}}

### Second Chance

[Second Chance](/features/second-chance/) emails are sent 1 hour and 24 hours after orders are created. By default, the order status changes from **Pending payment** to **Cancelled** after 8 hours (480 minutes).

If the customer pays via the **second** email (24 hours later), the payment is processed but the transaction update may not be handled correctly in Magento 2 because the order has expired. This may cause issues with external services, e.g. ERP/inventory management, if items are low in stock, or for one-off products like antiques.

To avoid this, match the order lifetime to the [payment link](/payment-links/) lifetime.

{{< details title="Setting order lifetimes">}}

1. Sign in to your Magento 2 backend.
2. Go to **Stores** > **Configuration** > **Sales** > **Sales** > **Orders cron settings**.
3. Select **Pending payment order lifetime (minutes)** and set to **2880 minutes**.

> **Note:** We recommend setting order lifetimes to 2 days (2880 minutes) to allow enough time for the customer to pay, but avoid issues with external services.

Reviewer: This is partially duplicating "Order lifetimes" earlier in the page. Maybe we should reduce it to just the note and link to that earlier section?

Author: Ah yes, I think I meant to come back and do something about that. Thanks!

> {{< details title="Setting order lifetimes">}}
> 
> 1. Sign in to your Magento 2 backend.
> 2. Go to **Stores** > **Configuration** > **Sales** > **Sales** > **Orders cron settings**.
> 3. Select **Pending payment order lifetime (minutes)** and set to **2880 minutes**.
> 
> **Note:** We recommend setting order lifetimes to 2 days (2880 minutes) to allow enough time for the customer to pay, but avoid issues with external services.
> 
> {{< /details >}}

Author: ```suggestion
See [Setting order lifetimes](#order-lifetimes) above. 

**Note:** We recommend setting order lifetimes to 2 days (2880 minutes) to allow enough time for the customer to pay, but avoid issues with external services.
```

{{< details title="Setting payment link lifetimes">}}

1. Sign in to your Magento 2 backend.
2. Go to **Stores** > **Configuration** > **MultiSafepay** > **Payment gateways**, and select the relevant gateway
3. Go to **Custom payment link lifetime**, select **Pending payment order lifetime**, and set to **2880 minutes**.

{{< /details >}}

### Shipping orders

When you ship [pay later](/payment-methods/pay-later/) orders, you need to change the order status from **Completed** to **Shipped**. This prevents the order expiring and triggers invoicing. 

If you do so in your Magento 2 backend, the updated status is passed to your MultiSafepay dashboard automatically.

### Surcharges

> [Surcharges](/surcharges/) are no longer supported.

Reviewer: Should we put this in an alert notice to make this stand out more?

Author: I don't think so, because this has been the case for a long time now. I'll adjust to say simply they're not supported. Good feedback though.

> [Surcharges](/surcharges/) are no longer supported.

Author: ```suggestion
[Surcharges](/surcharges/) are not supported.
```

> {{< details title="Using Fooman for surcharges" >}}

Reviewer: ```suggestion
{{< details title="Using Fooman for surcharges (Deprecated)" >}}
```
Since we don't support surcharges anymore?

Author: No, our plugin doesn't support surcharges, but merchants can use Fooman to apply surcharges.

    - Clear your cache.

{{< /details >}}

### Upgrading 
If you are still using the [deprecated plugin](https://github.com/MultiSafepay/Magento2Msp), we recommend upgrading to the latest version as soon as possible.

{{< details title="Why upgrade?" >}}

The new plugin features code improvements, and unit and integration testing. It is built on top of the Magento payment provider gateway structure. Some payment methods now skip the MultiSafepay payment page, which increases conversion.

We support most Magento functionalities. For any questions, email <integration@multisafepay.com>

**New features**

- Improved:
    - Magento backend configuration, including support information
    - Translations
    - Error handling, and event and error logs
- Documentation for payment methods 
- Modular setup for greater installation flexibility
- PWA (GraphQL) endpoints
- Support for [Magento Vault](https://devdocs.magento.com/guides/v2.4/payments-integrations/vault/vault-intro.html) and [Instant Purchase](https://docs.magento.com/user-guide/sales/checkout-instant-purchase.html) (replace [recurring payments](/features/recurring-payments/))

**Configuration fields**

> We have removed or altered the following configuration fields. 
> 

Author: ```suggestion
We have removed or altered the following configuration fields:
```

> If you want a field from the deprecated plugin back, email <integration@multisafepay.com>

Author: ```suggestion
```

> _Emailing invoices to customers_

Reviewer: Maybe we should add a subtitle here to help separate the content above and below? Something like: "Changes"?
